tests/tests_pytorch/accelerators/test_ipu.py

```diff
@@ -40,16 +40,19 @@
 
 class IPUModel(BoringModel):
     def training_step(self, batch, batch_idx):
+        assert self.precision == torch.finfo(batch.dtype).bits
```

Comment:
These lines would fail without the fix in the strategy, correct?

Reply:
It passes now, but I haven't verified that it would have failed before. We shouldn't merge until I have

Reply:
Since `batch_to_device` is the last thing that happens before the tensors are _supposed_ to be on the device, the updated test should be valid. It fails before the PR and passes after.